file_store/src/lora_valid_poc.rs

```diff
@@ -53,12 +54,34 @@ impl TryFrom<LoraValidPocV1> for LoraValidPoc {
     }
 }
 
+impl TryFrom<LoraValidPoc> for LoraValidPocV1 {
+    type Error = Error;
+    fn try_from(v: LoraValidPoc) -> Result<Self> {
+        let beacon_report: LoraValidBeaconReportV1 = v.beacon_report.try_into()?;
+        let mut witnesses: Vec<LoraValidWitnessReportV1> = Vec::new();
+        for witness_proto in v.witness_reports {
+            let witness_report: LoraValidWitnessReportV1 = witness_proto.try_into()?;
+            witnesses.push(witness_report)
+        }
+        Ok(Self {
+            poc_id: v.poc_id,
+            beacon_report: Some(beacon_report),
+            witness_reports: witnesses,
+        })
+    }
+}
+
 impl TryFrom<LoraValidBeaconReportV1> for LoraValidBeaconReport {
     type Error = Error;
     fn try_from(v: LoraValidBeaconReportV1) -> Result<Self> {
+        let location = v
+            .location
+            .is_empty()
+            .not()
+            .then(|| v.location.parse::<u64>().unwrap());
```

Comment:
is this saying "if location is not empty then parse it as a u64"? what happens if the field _is_ empty? since the struct definition above is setting the location field to an `Option` i think the rust-y way to do this is to "assume a value is present and parsable and then handle the alternative if it isn't"; i believe this is equivalent and a little cleaner
```suggestion
        let location: Option<u64> = v.location.parse().ok();
```
or alternatively `let location = v.location.parse::<u64>().ok();` if you prefer.
the type annotation makes the assumption that the value will parse to a u64 and the `ok()` will convert the `Result(u64)` to a `Some(u64)` or else a `None` if the value is empty

Reply:
the `ok()` is actually even more forgiving, so _any_ error in parsing the string to a u64 will result in `None` meaning `"".parse()` and `"foo".parse()` will both be converted to `None`

Reply:
if the proto location is an empty string then the struct value will be None.  If its not empty then the string will be parsed to an int.  Also has a suggestion from madninja to get rid of the unwrap and use transpose on the outer.  I plan on going over all the unwrap usage n a sep pass once i have ev thing functional